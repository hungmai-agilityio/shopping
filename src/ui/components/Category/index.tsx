'use client';

import { useState } from 'react';
import { clsx } from 'clsx';
import { useRouter, useSearchParams } from 'next/navigation';

// Constants
import { mada } from '@/constants';

interface CategoryProps {
  categories: string[];
  queryParam: string;
}

export const Category = ({ categories, queryParam }: CategoryProps) => {
  const { push } = useRouter();
  const searchParams = useSearchParams();
  const params = new URLSearchParams(searchParams.toString());
  const initCategory = searchParams.get(queryParam) || '';

  const [currentCategory, setCurrentCategory] = useState<string>(initCategory);

  // Update category and URL when a new category is selected
  const handleCategorySelect = (category: string) => {
    params.set(queryParam, category);
    push(`?${params.toString()}`, { scroll: false });
    setCurrentCategory(category);
  };

  return (
    <>
      <ul
        className={`${mada.className} my-5 flex xl:gap-10 gap-5 justify-center text-dark`}
      >
        {categories.map((category, index) => (
          <li
            className={clsx(
              'relative capitalize cursor-pointer hover:text-second',
              {
                'text-second font-bold': currentCategory === category
              }
            )}
            key={index}
            onClick={handleCategorySelect.bind(null, category)}
          >
            {category}
            {currentCategory === category && (
              <span className="absolute left-0 right-0 -bottom-2 mx-auto h-line w-full bg-active"></span>
            )}
          </li>
        ))}
      </ul>
    </>
  );
};
