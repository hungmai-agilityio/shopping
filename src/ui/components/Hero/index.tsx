import { ReactNode } from 'react';
import { clsx } from 'clsx';

// Components
import Breadcrumb, { BreadcrumbItem } from '@/ui/components/common/BreadCrumb';

interface HeroProps {
  isHeroSecond?: boolean;
  isBreadCrumb?: boolean;
  styles?: string;
  children: ReactNode;
  items?: BreadcrumbItem[];
}

const Hero = ({
  isHeroSecond = false,
  isBreadCrumb,
  styles,
  children,
  items
}: HeroProps) => {
  return (
    <div
      className={clsx(
        'relative w-full h-full overflow-hidden py-10 px-28',
        styles
      )}
    >
      {isBreadCrumb && (
        <Breadcrumb
          items={items!}
          color="text-white"
          styles="relative container z-20 md:flex hidden"
        />
      )}
      <div className="relative z-10">{children}</div>

      <div
        className={clsx('absolute inset-0 bg-gradient-primary ', {
          'top-0 left-0 right-0 bottom-0 -skew-y-6 origin-top-left max-h-hero':
            isHeroSecond
        })}
      ></div>
    </div>
  );
};

export default Hero;
