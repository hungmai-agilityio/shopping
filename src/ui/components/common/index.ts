export { default as BreadCrumb } from '@/ui/components/common/BreadCrumb';
export { default as Button } from '@/ui/components/common/Button';
export { default as Badge } from '@/ui/components/common/Badge';
export { default as RadioButton } from '@/ui/components/common/RadioButton';
export { default as Icon } from '@/ui/components/common/Icon';
export { default as Input } from '@/ui/components/common/Input';
export { default as InputController } from '@/ui/components/common/Input/InputController';
export { default as Logo } from '@/ui/components/common/Logo';
export { default as Table } from '@/ui/components/common/Table';
export { default as Typography } from '@/ui/components/common/Typography';
export { default as Quantity } from '@/ui/components/common/Quantity';
export { default as ColorPicker } from '@/ui/components/common/ColorPicker';
export { default as ToastMessage } from '@/ui/components/common/ToastMessage';
export { default as Spinner } from '@/ui/components/common/Loading/Spinner'
export { default as Brand } from '@/ui/components/common/Brand'
export { default as Modal } from '@/ui/components/common/Modal';
export { default as ModalDelete } from '@/ui/components/common/Modal/Delete';
export { default as ModalAuth } from '@/ui/components/common/Modal/Auth';
export { default as Dropdown } from '@/ui/components/common/Dropdown';
export { default as DropdownNote } from '@/ui/components/common/Dropdown/Note';
export { default as CardProductSkeleton } from '@/ui/components/common/Skeleton/CardProduct';
export { default as Carousel } from '@/ui/components/common/Carousel';
export { default as CarouselCategories } from '@/ui/components/common/Carousel/Categories';
export { default as CarouselProduct } from '@/ui/components/common/Carousel/Product';
export { default as BurgerMenu } from '@/ui/components/common/BurgerMenu';
export { default as CardImage } from '@/ui/components/common/Card/Image';
export { default as CardProduct } from '@/ui/components/common/Card/Product';
export { default as CardProfile } from '@/ui/components/common/Card/Profile';
export { default as Card } from '@/ui/components/common/Card';
export { default as List } from '@/ui/components/common/List';
export { default as Navbar } from '@/ui/components/common/Navbar';

export * from '@/ui/components/common/Table/Body';
export * from '@/ui/components/common/Table/Row';
export * from '@/ui/components/common/Table/Row/Empty';
export * from '@/ui/components/common/Table/Cell';
export * from '@/ui/components/common/Table/Head';
export * from '@/ui/components/common/Typography/Heading';
