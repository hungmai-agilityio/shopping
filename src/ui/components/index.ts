export { default as UserAction } from '@/ui/components/UserAction';
export { default as UserIcon } from '@/ui/components/UserAction/Icon';
export { default as CartProduct } from '@/ui/components/Cart/Product';
export { default as CartItem } from '@/ui/components/Cart/Product/Item';
export { default as CartCheckout } from '@/ui/components/Cart/Checkout';
export { default as AboutInfo } from '@/ui/components/AboutInfo';
export { default as AuthForm } from '@/ui/components/Auth';
export { default as WishList } from '@/ui/components/WishList';
export { default as Hero } from '@/ui/components/Hero';
export { default as Tabs } from '@/ui/components/common/Tabs';

export * from '@/ui/components/common';
export * from '@/ui/components/Category';
export * from '@/ui/components/common/Tabs/Panel';
