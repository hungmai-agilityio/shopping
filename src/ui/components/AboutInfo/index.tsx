import { FONT_SIZE, FONT_WEIGHT, mada, TAG } from '@/constants';
import CardImage from '@/ui/components/Card/Image';
import { Typography } from '@/ui/components/common';
import { clsx } from 'clsx';

interface AboutInfoProps {
  desc: string;
  isReverse?: boolean;
  aboutName: string;
  src: string;
  title: string;
  styles?: string;
}

const AboutInfo = ({
  title,
  desc,
  aboutName,
  src,
  isReverse,
  styles
}: AboutInfoProps) => {
  return (
    <div
      className={clsx(
        'lg:flex block justify-between w-full gap-10',
        {
          'flex-row-reverse': isReverse
        },
        styles
      )}
    >
      <div className="lg:max-w-card-sm lg:text-left text-center">
        <Typography
          fontWeight={FONT_WEIGHT.BOLD}
          tag={TAG.H2}
          size={FONT_SIZE.MEDIUM}
          className={`${mada.className} capitalize my-5`}
        >
          {title}
        </Typography>
        <Typography className="my-6 text-dark-md" size={FONT_SIZE.X_SMALL}>
          {desc}
        </Typography>
      </div>
      <div className="flex justify-center">
        <CardImage
          src={src}
          alt={aboutName}
          width={590}
          height={349}
          title={aboutName}
          isTitleLeft={isReverse}
          className="w-full h-96"
        />
      </div>
    </div>
  );
};

export default AboutInfo;
