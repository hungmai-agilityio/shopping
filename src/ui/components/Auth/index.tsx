import { ReactNode } from 'react';
import { BreadCrumb, Heading, Typography } from '@/ui/components/common';
import { FONT_SIZE, FONT_WEIGHT, TAG, mada, popping } from '@/constants';

interface AuthFormProps {
  title: string;
  subtitle: ReactNode;
  backgroundImage: string;
  children: ReactNode;
}

const AuthForm = ({
  title,
  subtitle,
  backgroundImage,
  children
}: AuthFormProps) => {
  const breadCrumb = [
    { label: 'Home', href: '/' },
    { label: 'My Account', href: '' }
  ];

  return (
    <section
      className={`${popping.className} bg-no-repeat bg-cover text-white`}
      style={{ backgroundImage: `url(${backgroundImage})` }}
    >
      <div className="bg-gradient-overlay py-20">
        <div className="w-full container">
          <div className="md:max-w-card-sm">
            <BreadCrumb items={breadCrumb} color="text-white" />
            <Heading>My Account</Heading>
            <Typography
              color="text-white"
              fontWeight={FONT_WEIGHT.BOLD}
              tag={TAG.H2}
              size={FONT_SIZE.MEDIUM}
              className={`${mada.className} mt-11`}
            >
              {title}
            </Typography>
            <div className="mt-6">{subtitle}</div>
            <div className="mt-10 mb-20">{children}</div>
          </div>
        </div>
      </div>
    </section>
  );
};

export default AuthForm;
