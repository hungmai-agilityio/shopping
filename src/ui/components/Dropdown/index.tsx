'use client';

import { ReactNode, useRef, useState } from 'react';
import { clsx } from 'clsx';

import { Icon } from '@/ui/components/common';
import { useOutsideClick } from '@/hooks';

interface DropdownProps {
  children: ReactNode;
  styles?: string;
  setIsNoteVisible?: (visible: boolean) => void;
  isNoteVisible?: boolean;
}
const Dropdown = ({
  children,
  styles,
  setIsNoteVisible,
  isNoteVisible
}: DropdownProps) => {
  const dropdownRef = useRef<HTMLDivElement>(null);

  const toggleNoteVisibility = () =>
    setIsNoteVisible && setIsNoteVisible(!isNoteVisible);

  useOutsideClick({
    ref: dropdownRef,
    handler: () => setIsNoteVisible && setIsNoteVisible(false)
  });

  return (
    <div ref={dropdownRef}>
      <div
        className={clsx('flex gap-5 items-center', styles)}
        onClick={toggleNoteVisibility}
      >
        <p className="text-second text-lg font-bold cursor-pointer lowercase">
          products notes
        </p>
        <Icon src="/arrow.svg" alt="arrow icon" width={24} height={24} />
      </div>
      {isNoteVisible && children}
    </div>
  );
};

export default Dropdown;
