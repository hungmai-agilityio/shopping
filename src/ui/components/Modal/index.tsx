import { Button } from '@/ui/components/common';
import { ReactNode } from 'react';

interface ModalProps {
  isOpen: boolean;
  onClose: () => void;
  title?: string;
  children: ReactNode;
  isConfirm?: boolean;
  buttonName?: string;
  onConfirm?: () => void;
}

const Modal = ({
  isOpen,
  onClose,
  title,
  children,
  isConfirm,
  buttonName,
  onConfirm
}: ModalProps) => {
  if (!isOpen) return null;

  return (
    <div className="fixed inset-0 flex items-center justify-center z-50">
      <div
        className="fixed inset-0 bg-black opacity-50"
        onClick={onClose}
      ></div>
      <div className="bg-white rounded-lg shadow-lg max-w-md w-full z-10">
        <div className="p-4 border-b">
          <h2 className="text-lg font-bold">{title}</h2>
        </div>
        <div className="p-4">{children}</div>
        <div className="p-4 border-t flex justify-between">
          <Button
            className="bg-red-500 text-white py-2 px-4 rounded"
            onClick={onClose}
          >
            Close
          </Button>
          {isConfirm && <Button onClick={onConfirm}>{buttonName}</Button>}
        </div>
      </div>
    </div>
  );
};

export default Modal;
