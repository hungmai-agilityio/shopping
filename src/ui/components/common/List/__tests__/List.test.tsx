import { render, screen } from '@testing-library/react';

// Constants
import { itemsHome } from '@/constants';

// Components
import { List } from '@/ui/components';

describe('List component', () => {
  const defaultProps = {
    heading: 'Home',
    color: 'text-white',
    items: itemsHome
  };
  it('render list with default props', () => {
    render(<List {...defaultProps} />);

    expect(screen.getByText('Home')).toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const { container } = render(<List {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
