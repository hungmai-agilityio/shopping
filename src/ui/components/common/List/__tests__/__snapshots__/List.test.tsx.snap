// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component Should match snapshot 1`] = `
<div>
  <div
    class="className"
  >
    <h4
      class="font-bold text-white lg:text-lg text-sm capitalize"
    >
      Home
    </h4>
    <div
      class="mt-5"
    >
      <a
        class="text-white block mb-4 capitalize text-sm cursor-pointer hover:text-purple-700"
        href="/"
      >
        Product
      </a>
      <a
        class="text-white block mb-4 capitalize text-sm cursor-pointer hover:text-purple-700"
        href="/"
      >
        Categories
      </a>
      <a
        class="text-white block mb-4 capitalize text-sm cursor-pointer hover:text-purple-700"
        href="/signIn"
      >
        Login
      </a>
    </div>
  </div>
</div>
`;
